gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -826,6 +813,11 @@ function roleselection.SelectRoles(plys, maxPlys)
 
 	plys = roleselection.GetSelectablePlayers(plys or player.GetAll())
 
+	-- we should give the player list an inital shuffle,
+	-- because if we're populating from player.GetAll(),
+	-- it heavily biases players that connected first for roles
+	table.Shuffle(plys)
```

Comment:
Could you please explain this?
As i can currently not make out any part of the code below where this is the case since the players are selected at random for each role. The table order is in no way considered for the role selection, at least as far as i can see this currently.

Reply:
I think its fine, it was just moved up. They removed the math.random pick in SelectBaseRolePlayers and shuffled the list instead.

So my suggestion was to simplify the comment. "Randomize role assignment by shuffling the list early"